File: src/main/java/org/nanowrimo/ws/UserWordCount.java
package org.nanowrimo.ws;

/**
 * <p>A summary of a given user - name, numeric Nanowrimo user ID, and their current wordcount.</p>
 * <p/>
 * <p>The <i>entries</i> property of a <code>UserWordCount</code> is filled in when retrieving their
 * history and will be <code>null</code> when retrieving the user's wordcount summary.</p>
 */
public class UserWordCount extends WordCountSummary {
    private WordCountListEntry[] entries;

    public UserWordCount() {
    }

    public int getUserID() {
        return getSummaryID();
    }

    public void setUserID(int userID) {
        setSummaryID(userID);
    }

    public WordCountListEntry[] getEntries() {
        return entries;
    }

    public void setEntries(WordCountListEntry[] entries) {
        this.entries = entries;
    }
}

File: src/main/java/org/nanowrimo/ws/UserWordCount.kt
package org.nanowrimo.ws

/**
 * A summary of a given user - name, numeric Nanowrimo user ID, and their current wordcount.
 *
 * The *entries* property of a `UserWordCount` is filled in when retrieving their
 * history and will be empty when retrieving the user's wordcount summary.
 */
class UserWordCount(var entries: List<WordCountListEntry>? = listOf()) : WordCountSummary() {
    var userID: Int
        get() = summaryID
        set(value) {
            summaryID = value
        }
}
